### PR TITLE
Fix server processing of an empty `data-wp-context` directive

### PR DESCRIPTION
**Pull request by luisherranz**
## What?

This is a follow up on https://github.com/WordPress/gutenberg/pull/55436.

The previous PR fixed the problem when the JSON provided to `data-wp-context` was not valid, and this PR fixes the problem when the `data-wp-context` is empty.

## Why?

Because, as explained by @dmsnell in https://github.com/WordPress/gutenberg/pull/55368, the user might forget the enter any value:

```html
<div data-wp-context>...</div>
```

## How?

I changed the logic a little bit to accommodate both checks (invalid JSON and empty value).

## Testing Instructions

This is covered by a new unit test, but if you want to test it out:

- Add an interactive block.
- Add a div with a `data-wp-context` directive and valid JSON.
- Add a nested div inside with an empty `data-wp-context`.
- Check that it doesn't trigger a fatal error and that the initial context is still valid.

### CODE REVIEW

**Review by dmsnell — APPROVED** [review]
It could be helpful to split the test into two since we're testing two separate situations. In fact, we already have a separate test for malformed JSON, so it could fit nicely to have one for "empty_directive" and one for "directive_without_value"

Either way this is an improvement so it would be good to get in, but it would be nice to split these here or in a follow-up PR.

**Comment by luisherranz** [comment]
What do you mean by "empty_directive" and "directive_without_value"? What's the difference between those?

**Comment by dmsnell** [comment]
> What do you mean by "empty_directive" and "directive_without_value"? What's the difference between those?

`data-wp-context=""` is an empty attribute and `data-wp-context` is a boolean attribute, one without a value. these are two distinct constructions in HTML and the HTML API exposes that difference.

in the DOM they both carry the value `""` but in the HTML API they will be `""` and `true`, respectively. this is something that I find unfortunate but don't know how to resolve, because `empty( $processor->get_attribute( … ) )` can return false for the boolean attributes, since `empty( true )` is `false`.

**Comment by luisherranz** [comment]
Oh, ok. That makes sense, thanks Dennis 🙂

I've added a new test for that use case. I'll wait for your confirmation before merging the PR.

**Comment by dmsnell** [comment]
> confirmation before merging the PR.

good to go!

**Pull request merged**
